packages/astro/src/core/build/css-asset-name.ts

```diff
@@ -24,8 +25,13 @@ export function shortHashedName(id: string, ctx: { getModuleInfo: GetModuleInfo
 export function createNameHash(baseId: string | undefined, hashIds: string[]): string {
 	const baseName = baseId ? prettifyBaseName(npath.parse(baseId).name) : 'index';
 	const hash = crypto.createHash('sha256');
+	const cwd = process.cwd();
+
 	for (const id of hashIds) {
-		hash.update(id, 'utf-8');
+		// Strip the project directory from the paths before they are hashed, so that assets
+		// that import these css files have consistent hashes when built in different environments.
+		const relativePath = npath.relative(cwd, id);
+		hash.update(relativePath, 'utf-8');
```

Comment:
I think it'd be more accurate if it's relative from the project root instead of cwd. You should be able to get that from `settings.config.root` in `plugin-css.ts` which uses this function.

Also, the slashes could be backslashes on Windows. We could use `normalizePath()` from `vite` to make sure that the slashes are always forward slashes.